### PR TITLE
Fix mso_schema_site_vrf_region_cidr and add VGW attribute to mso_schema_site_vrf_region_cidr_subnet

### LINKED ISSUE #50: [mso_schema_site_vrf_region_cidr_subnet] Add Azure "Used by VGW" support
On an Azure Site, at least one subnet of a CIDR need to be marked for "Used by VGW" to complete a proper deployment.
This request ask to add this support in the mso_schema_site_vrf_region_cidr_subnet module.

### LINKED ISSUE #66: [mso_schema_template_anp_epg] AttributeError: 'NoneType' object has no attribute 'group'
Module Failure when using state [query, present]

AttributeError: 'NoneType' object has no attribute 'group'

[mso_schema_template_anp_epg.error.txt](https://github.com/CiscoDevNet/ansible-mso/files/4845788/mso_schema_template_anp_epg.error.txt)

### PR DESCRIPTION
**Pull request by lhercot**
Fix mso_schema_site_vrf_region_cidr to automatically create VRF and Region if not present at site level
Add support for VGW attribute in mso_schema_site_vrf_region_cidr_subnet
Remove unused regions attribute from mso_schema_template_vrf
Add test for mso_schema_site_vrf_region_cidr module
Add test for mso_schema_site_vrf_region_cidr_subnet module

Resolves #50 
Resolves #66

### CODE REVIEW

File: plugins/modules/mso_schema_site_vrf_region_cidr.py [hunk 1]
```diff
@@ -178,48 +181,93 @@ def main():
...
+                    cidr_idx = cidrs.index(cidr)
+                    # FIXME: Changes based on index are DANGEROUS
+                    cidr_path = '/sites/{0}/vrfs/{1}/regions/{2}/cidrs/{3}'.format(site_template, vrf, region, cidr_idx)
+                    mso.existing = schema_obj.get('sites')[site_idx]['vrfs'][vrf_idx]['regions'][region_idx]['cidrs'][cidr_idx]
+                op_path = '/sites/{0}/vrfs/{1}/regions/{2}/cidrs/-'.format(site_template, vrf, region)
+                payload = new_cidr
 
     if state == 'query':
-        if cidr is None:
+        if region not in regions:
```
Comment (xinyuezhao) [inline plugins/modules/mso_schema_site_vrf_region_cidr.py:263]: what's error msg if vrf doesn't exist at site level but exist at template level?

Comment (lhercot) [reply plugins/modules/mso_schema_site_vrf_region_cidr.py:263]: No error message, the VRF will be automatically created at VRF level see code from line 223 to 235.

Comment (xinyuezhao) [reply plugins/modules/mso_schema_site_vrf_region_cidr.py:263]: In my understanding, we do create region automatically while it doesn't exist in site level. But when we do query, we give an error msg when there is no site-level region. 
should we do the same in vrf?

File: tests/integration/targets/mso_schema_template_vrf_contract/tasks/main.yml [hunk 2]
```diff
@@ -41,6 +41,15 @@
 #     - https://{{ apic_hostname }}
 #     state: present
```
Comment (shrsr) [inline tests/integration/targets/mso_schema_template_vrf_contract/tasks/main.yml:43]: The Remove Schemas block has been shifted. What's the reason?

Comment (lhercot) [reply tests/integration/targets/mso_schema_template_vrf_contract/tasks/main.yml:43]: As discussed this is required to make it work for the new cloud modules.

**Pull request merged**